### PR TITLE
"Parodus crash due to invalid serial number"

### PR DESCRIPTION
**Pull request by vasuki01**

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -208,6 +208,29 @@ int parse_mac_address (char *target, const char *arg)
 	return 0;
 }
 
+int parse_serial_num(char *target, const char *arg)
+{
+	char ch;
+	if(strlen(arg) == 0)
```
Comment (sadhyama) [inline src/config.c:214]: add NULL check and then do strlen to avoid crash

File: src/config.c [hunk 2]
```diff
@@ -208,6 +208,29 @@ int parse_mac_address (char *target, const char *arg)
...
+	{
+	    ParodusError("Empty serial number\n");
+	    return -1;
+	}
+	for(int i=0; (ch = arg[i]) != '\0'; i++)
+	{
+		if((ch >= 97 && ch <= 122) || (ch >= 65 && ch <= 90) || (ch >=48 && ch <= 57))
```
Comment (sadhyama) [inline src/config.c:221]: Add validation comment here which all valid values are included , A-Z, a-z etc

File: src/config.c [hunk 3]
```diff
@@ -462,8 +485,12 @@ int parseCommandLine(int argc,char **argv,ParodusCfg * cfg)
          break;
         
         case 's':
-          parStrncpy(cfg->hw_serial_number,optarg,sizeof(cfg->hw_serial_number));
-          ParodusInfo("hw_serial_number is %s\n",cfg->hw_serial_number);
+	if(parse_serial_num(cfg->hw_serial_number, optarg) == 0) {
+            ParodusInfo ("hw_serial-number is %s\n",cfg->hw_serial_number);
+					} else {
+						ParodusError ("Bad serial-number %s\n", optarg);
+						return -1;
```
Comment (sadhyama) [inline src/config.c:492]: I think return is not required as serial number is not a mandatory argument. we should just log an error and proceed with other arguments

**Review by sadhyama — CHANGES_REQUESTED** [review]

File: src/config.c [hunk 4]
```diff
@@ -208,6 +208,29 @@ int parse_mac_address (char *target, const char *arg)
 	return 0;
 }
 
+int parse_serial_num(char *target, const char *arg)
+{
+	char ch;
+	if(strlen(arg) == 0)
+	{
+	    ParodusError("Empty serial number\n");
+	    return -1;
```
Comment (shilpa24balaji) [inline src/config.c:217]: Lets not return error when it is empty, just log error and set to default "unknown"

File: src/config.c (hunk 3 above)
Comment (shilpa24balaji) [reply src/config.c:492]: For junk value lets log error and set to default unknown as its not mandatory

**Pull request merged**